linkerd2.yaml

```diff
@@ -0,0 +1,209 @@
+package:
+  name: linkerd2
+  version: 24.10.3
+  epoch: 0
+  description: "meta linkerd package"
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - cargo-auditable
+      - clang
+      - cmake
+      - nodejs
+      - openssf-compiler-options
+      - openssl-dev
+      - perl-dev
+      - rust
+      - yarn
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 64130e1eb5313a4b15c5b279edeb6f263b70fdd5
+      repository: https://github.com/linkerd/linkerd2/
+      tag: edge-${{package.version}}
+
+subpackages:
+  - name: ${{package.name}}-web
+    pipeline:
+      - runs: |
+          ./bin/web setup install --frozen-lockfile
+          ./bin/web build
+          go build -mod=readonly -o web/web -ldflags "-w" ./web
+          mkdir -p ${{targets.contextdir}}/app
+          mkdir -p ${{targets.contextdir}}/templates
+          cp -r ./web/app/dist ${{targets.contextdir}}/app
+          cp -r ./web/templates ${{targets.contextdir}}/
+          cp ./web/web ${{targets.contextdir}}/
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}-web
+      pipeline:
+        - name: version and help message
+          runs: |
+            /web -version
+            /web -help
+            /web 2>&1 | grep 'starting admin server'
+
+  - name: ${{package.name}}-cli
+    pipeline:
+      - runs: |
+          go generate -mod=readonly ./pkg/charts/static
+          go generate -mod=readonly ./jaeger/static
+          go generate -mod=readonly ./multicluster/static
+          go generate -mod=readonly ./viz/static
+      - uses: go/build
+        with:
+          packages: ./cli
+          tags: prod
+          output: linkerd
+          ldflags: -X github.com/linkerd/linkerd2/pkg/version.Version=edge-${{package.version}}
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}-cli
+      pipeline:
+        - name: linkerd version check
+          runs: linkerd version --client
+
+  - name: ${{package.name}}-proxy-identity
+    pipeline:
+      - uses: go/build
+        with:
+          packages: ./proxy-identity/main.go
+          output: linkerd2-proxy-identity
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/lib/linkerd
+          mv ${{targets.contextdir}}/usr/bin/linkerd2-proxy-identity ${{targets.contextdir}}/usr/lib/linkerd
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}-proxy-identity
```

Comment:
We don't need to include the subpackage being tested in subpackage test pipelines